### ISSUE TITLE
 add more cue+bin format support

### ISSUE THREAD

**Issue by crashGG** [flyinghead/flycast#434]
The current flycast does not seem to support mode 2 tracks

here is my biova.cue :

FILE "biova.bin" BINARY
  REM SESSION 01
    TRACK 01 AUDIO
      INDEX 01 00:00:00
         REM LBA: 0
  REM LEAD-OUT 07:28:00
  REM SESSION 02
    TRACK 02 MODE2/2352
      INDEX 01 10:00:00
         REM LBA: 45000

this biova.cue can successfully loaded and run with DEMUL. but with flycast it shows:
<img width="582" alt="2021-12-04_103800" src="https://user-images.githubusercontent.com/80488411/144701969-60340e11-4873-4cf7-a55a-d43feaec07b5.png">

In addition, flycast does not seem to recognize the CHD converted from cue+bin,only CHD converted from gdi+bin works fine.(All image files meet the "redump" specification)

**Comment by cinnamonmatexfce** [comment]
> In addition, flycast does not seem to recognize the CHD converted from cue+bin,only CHD converted from gdi+bin works fine.(All image files meet the "redump" specification)

https://github.com/mamedev/mame/issues/6466

**Comment by bslenul** [comment]
Yeah CHD conversion from Redump was added almost a year ago: https://github.com/mamedev/mame/pull/7422

You just have to update your chdman version (at least 0.227 IIRC). I believe it's still not possible to convert DC games from CHD back to bin/cue however.

**Comment by crashGG** [comment]
> Yeah CHD conversion from Redump was added almost a year ago: [mamedev/mame#7422](https://github.com/mamedev/mame/pull/7422)
> 
> You just have to update your chdman version (at least 0.227 IIRC). I believe it's still not possible to convert DC games from CHD back to bin/cue however. **edit:** Oof yeah: [mamedev/mame#7717](https://github.com/mamedev/mame/pull/7717) looks like it won't happen anytime soon... 😅 But at least bin/cue -> chd works fine.

Thank you, you are right. After your reminder, I have updated the latest chdman. Here what I mainly want to express is that flycast does not currently support MODE2/2352 track.

**Comment by flyinghead** [comment]
MODE2/2352 tracks are supported. The problem is your .cue file is using an old multi-session Isobuster format that isn't supported by Flycast. 

More details here: http://forum.redump.org/post/69640/#p69640

**Comment by crashGG** [comment]
> MODE2/2352 tracks are supported. The problem is your .cue file is using an old multi-session Isobuster format that isn't supported by Flycast.
> 
> More details here: http://forum.redump.org/post/69640/#p69640

In my current situation, how should I change my cue code? In fact,I simplified it as follows：

FILE "biova.bin" BINARY
REM SESSION 01
  TRACK 01 MODE2/2352
    INDEX 01 00:00:00
REM SESSION 02
  TRACK 02 MODE2/2352
    INDEX 01 10:00:00

 , demul can also run the game with this cue. Even change this cue converted to chd, demul can also run.
I mainly face the following situation: there are many translated DC games that use the cdi format. This format cannot be directly converted to chd. So I first convert cdi to cue+bin using ultraiso. Then convert cue+bin to chd through chdman. The final chd format image can be run by demul. You can also try it to see where the problem is. There are many dc mirror downloads in cdi format on the Internet. For example: https://cdromance.com/dc-iso/frame-gride-english-patched/

By the way, flycast does not support chd with mode2/2352 format
In chd.cpp of flycast , it seems that it does not support mode2/2352, only mode1/2352:
https://github.com/libretro/flycast/blob/master/core/imgread/chd.cpp

sincerely hope that flycast will become more and more perfect :)

**Comment by flyinghead** [comment]
This .CUE uses a single file for all tracks. Flycast currently only supports one file per track.

May I ask why you need to convert CDI to CHD? Why not using the CDI format directly? There's nothing wrong with it.

**Comment by crashGG** [comment]
Because many cdi images are added with dummy files to expand the capacity when they are made. These files have a high degree of redundancy and can be highly compressed. In the current simulator environment, compressing the image file to chd can save disk space. In addition, the chd file itself also has a checksum code. It is easy to detect whether it has been tampered with or damaged. It is more reliable than storing cdi files directly.

**Comment by frafes** [comment]
I second this request... it would be great if this were added.
